### ISSUE TITLE
Audio files not generating with a shared Anki deck unless in very small batches

### ISSUE THREAD

**Issue by darrenhaken** [Vocab-Apps/anki-hyper-tts#203]
I've been trying to apply audio files to the following deck:
https://drive.google.com/file/d/1sXlEIOM5rTyhwS-Xv51qy3HDyYaFLe1a/view

It is a Refold deck with ~16k for Dutch.

Applying Hyper TTS to the entire corpus caused Anki to crash. I tried it in batches of 1000 and it didn't crash but it also didn't write the result to the Audios destination field. If I select 100 cards at a time, it works.

Here are my versions of Anki:
Version ⁨24.06.3 (d678e393)⁩
Python 3.9.18 Qt 6.6.2 PyQt 6.6.1

I have tried to apply this using Azure, ElevenLabs 2.0 and Elevenlabs 2.5 and they all fail.
Any ideas?

**Comment by luc-vocab** [comment]
Let's focus on reproducing the issue with a small number of notes. Don't try to mass-generate until you're satisfied with the output on say 5 notes. Is the audio field a separate field containing only audio ? What does the status on the generate screen in HyperTTS say for each note, does it say OK or does it say error ?
When you say :

> I tried it in batches of 1000 and it didn't crash but it also didn't write the result to the Audios destination field
How did you verify that audio was not generated ? What did you see in your browser screen ?

**Comment by darrenhaken** [comment]
@luc-vocab I managed to work out the issue. I was using a trial API key and I exceeded 5000 API calls. I did not see a notification for this in Anki.

I figured this out by using Google Translate with no issue. I then purchased a batch of API calls in the product and the entire corpus ran successfully, albeit taking about 8 hours.

I'll close the issue but it's worth being aware I didn't see an error about my API key.
It could be my error, I'm not sure.
